BuildingBlocks/src/BuildingBlocks.Infrastructure/EventBus/RabbitMQ/DefaultRabbitMQPersisterConnection.cs

```diff
@@ -86,27 +87,52 @@ public bool TryConnect()
     private void OnConnectionBlocked(object? sender, ConnectionBlockedEventArgs e)
     {
         if (_disposed) return;
-
-        _logger.LogWarning("A RabbitMQ connection is shutdown. Trying to re-connect...");
-
+        _logger.ConnectionIsOnShutdown();
         TryConnect();
     }
 
     private void OnCallbackException(object? sender, CallbackExceptionEventArgs e)
     {
         if (_disposed) return;
-
-        _logger.LogWarning("A RabbitMQ connection throw exception. Trying to re-connect...");
-
+        _logger.ConnectionThrewAnException();
         TryConnect();
     }
 
     private void OnConnectionShutdown(object? sender, ShutdownEventArgs reason)
     {
         if (_disposed) return;
-
-        _logger.LogWarning("A RabbitMQ connection is on shutdown. Trying to re-connect...");
-
+        _logger.ConnectionIsShutdown();
         TryConnect();
     }
 }
+
+internal static partial class DefaultRabbitMqPersistentConnectionLogs
+{
+    [LoggerMessage(
+        EventId = 715507,
+        EventName = "DefaultRabbitMqPersistentConnection.BrokerUnreachableException",
+        Level = LogLevel.Warning,
+        Message = "{exceptionString}")]
+    public static partial void BrokerUnreachableException(this ILogger logger, string exceptionString);
```

Comment:
Multiple suggestions:
```suggestion
    [LoggerMessage(
        EventId = 715507,
        EventName = "DefaultRabbitMqPersistentConnection.ConnectionError",
        Level = LogLevel.Warning,
        Message = "There was an error while trying to connect to RabbitMQ. Attemping to retry...")]
    public static partial void ConnectionError(this ILogger logger, Exception exception);
```